### PR TITLE
Add constructors to OperatorBase and all sub-classes

### PR DESCRIPTION
**Pull request by wangkuiyi**
Fixes https://github.com/PaddlePaddle/Paddle/issues/3429

### CODE REVIEW

File: paddle/framework/backward_test.cc [hunk 1]
```diff
@@ -30,6 +30,11 @@ using DeviceContext = platform::DeviceContext;
 
 class EmptyOp : public OperatorBase {
```
Comment (Canpio) [inline paddle/framework/backward_test.cc]: Shouldn't this be named `NOP`?

Comment (wangkuiyi) [inline paddle/framework/backward_test.cc]: Yes. We can create another PR to fix it.

Comment (Canpio) [inline paddle/framework/backward_test.cc]: Or we can fix it in following prs.

File: paddle/framework/grad_op_builder_test.cc [hunk 2]
```diff
@@ -10,6 +10,11 @@ namespace framework {
 
 class NOP : public OperatorBase {
  public:
+  NOP(const std::string &type, const std::vector<std::string> &inputs,
```
Comment (Canpio) [inline paddle/framework/grad_op_builder_test.cc]: It seems strange that NOP does nothing but need so much stuff to initialize it...

**Pull request merged**